cmake/Dependencies.cmake

```diff
@@ -221,47 +221,75 @@ if(USE_QNNPACK)
+      if (NOT DEFINED CPUINFO_SOURCE_DIR)
+        set(CPUINFO_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/cpuinfo" CACHE STRING "cpuinfo source directory")
+      endif()
+      if (NOT DEFINED FP16_SOURCE_DIR)
+        set(FP16_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/FP16" CACHE STRING "FP16 source directory")
+      endif()
+      if (NOT DEFINED FXDIV_SOURCE_DIR)
+        set(FXDIV_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/FXdiv" CACHE STRING "FXdiv source directory")
+      endif()
+      if (NOT DEFINED PSIMD_SOURCE_DIR)
+        set(PSIMD_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/psimd" CACHE STRING "PSimd source directory")
+      endif()
+      if (NOT DEFINED PTHREADPOOL_SOURCE_DIR)
+        set(PTHREADPOOL_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/pthreadpool" CACHE STRING "pthreadpool source directory")
+      endif()
 
-    if(NOT TARGET qnnpack)
-      set(QNNPACK_BUILD_TESTS OFF CACHE BOOL "")
-      set(QNNPACK_BUILD_BENCHMARKS OFF CACHE BOOL "")
-      set(QNNPACK_CUSTOM_THREADPOOL ON CACHE BOOL "")
-      set(QNNPACK_LIBRARY_TYPE "static" CACHE STRING "")
-      set(PTHREADPOOL_LIBRARY_TYPE "static" CACHE STRING "")
-      set(CPUINFO_LIBRARY_TYPE "static" CACHE STRING "")
-      set(CPUINFO_LOG_LEVEL "error" CACHE STRING "")
-      add_subdirectory(
-        "${QNNPACK_SOURCE_DIR}"
-        "${CONFU_DEPENDENCIES_BINARY_DIR}/QNNPACK")
-      # We build static versions of QNNPACK and pthreadpool but link
-      # them into a shared library for Caffe2, so they need PIC.
-      set_property(TARGET qnnpack PROPERTY POSITION_INDEPENDENT_CODE ON)
-      set_property(TARGET pthreadpool PROPERTY POSITION_INDEPENDENT_CODE ON)
-      set_property(TARGET cpuinfo PROPERTY POSITION_INDEPENDENT_CODE ON)
-    endif()
+    if (BUILD_CAFFE2_MOBILE)
+      if (NOT DEFINED QNNPACK_SOURCE_DIR)
+        set(QNNPACK_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/QNNPACK" CACHE STRING "QNNPACK source directory")
+      endif()
+
+      if(NOT TARGET qnnpack)
+        set(QNNPACK_BUILD_TESTS OFF CACHE BOOL "")
+        set(QNNPACK_BUILD_BENCHMARKS OFF CACHE BOOL "")
+        set(QNNPACK_CUSTOM_THREADPOOL ON CACHE BOOL "")
+        set(QNNPACK_LIBRARY_TYPE "static" CACHE STRING "")
+        set(PTHREADPOOL_LIBRARY_TYPE "static" CACHE STRING "")
+        set(CPUINFO_LIBRARY_TYPE "static" CACHE STRING "")
+        set(CPUINFO_LOG_LEVEL "error" CACHE STRING "")
+        add_subdirectory(
+          "${QNNPACK_SOURCE_DIR}"
+          "${CONFU_DEPENDENCIES_BINARY_DIR}/QNNPACK")
+        # We build static versions of QNNPACK and pthreadpool but link
+        # them into a shared library for Caffe2, so they need PIC.
+        set_property(TARGET qnnpack PROPERTY POSITION_INDEPENDENT_CODE ON)
+        set_property(TARGET pthreadpool PROPERTY POSITION_INDEPENDENT_CODE ON)
+        set_property(TARGET cpuinfo PROPERTY POSITION_INDEPENDENT_CODE ON)
+      endif()
+
+      list(APPEND Caffe2_DEPENDENCY_LIBS qnnpack)
+    else()
+      if (NOT DEFINED PYTORCH_QNNPACK_SOURCE_DIR)
```

Comment:
as I proposed, pytorch_qnnpack related logic can be put after qnnpack block, gated with USE_PYTORCH_QNNPACK macro.